TLM/TLM/LoadingExtension.cs

```diff
@@ -359,25 +355,30 @@ public class ManualHarmonyPatch {
                 Destroy<RemoveVehicleButtonExtender>();
                 Destroy<RemoveCitizenInstanceButtonExtender>();
 
+                // Custom path manger is destroyed when reloading. That is why the following code
+                // is commented out.
                 //simManager?.Remove(CustomPathManager);
-
                 //Object.Destroy(CustomPathManager);
                 //CustomPathManager = null;
 
+                if (TransportDemandUI != null) {
+                    UIView uiView = UIView.GetAView();
+                    Object.Destroy(TransportDemandUI);
+                    TransportDemandUI = null;
+                }
+
+                var tmTool = UIBase.GetTrafficManagerTool();
                 Log.Info("Removing Controls from UI.");
                 if (BaseUI != null) {
                     BaseUI.Close(); // Hide the UI ASAP
-                    Object.Destroy(BaseUI);
```

Comment:
why I don't see traffic manager in modtools?
![Screenshot (424)](https://user-images.githubusercontent.com/26344691/74087891-04405c80-4a99-11ea-8f97-c8b24473c0f0.png)

Sorry about the sloppy work. I got totally confused! you are spot on. I shall release trafficManger from UIBase destructor.

Reply:
OK this is fixed :) 
![Screenshot (425)](https://user-images.githubusercontent.com/26344691/74088039-a0b72e80-4a9a-11ea-8c2b-5d3eee1df0d0.png)
```
Debug 52.4193354: UIBase destructor is called.
Debug 52.4235121: LoadingExtension.DisableTool: called
Info 52.4301887: Removing Traffic Manager Tool.
Info 58.5976456: TrafficManagerTool.OnDestroy() called
Info 58.8597243: TM:PE disabled.
```


I cleaned up the UIBase destruction path a little bit. That and your suggestion fixed the problem. I think part of the problem was traffic manager destructor was never called because of some out of place null reference checks.